ouroboros-consensus-shelley-test/test/Test/ThreadNet/RealTPraos.hs

```diff
@@ -121,28 +204,71 @@ prop_simple_real_tpraos_convergence TestSetup
       , messageDelay = noCalcMessageDelay
       , nodeJoinPlan = trivialNodeJoinPlan numCoreNodes
       , nodeRestarts = noRestarts
-      , txGenExtra   = ShelleyTxGenExtra $ mkGenEnv coreNodes
+      , txGenExtra   = ShelleyTxGenExtra
+        { stgeGenEnv          = mkGenEnv inclPPUs coreNodes
+        , stgeStartAt         =
+            SlotNo $ if includingDUpdateTx then 1 else 0
+            -- We don't generate any transactions before the transaction
+            -- carrying the proposal because they might consume its inputs
+            -- before it does, thereby rendering it invalid.
+        }
       , version      = setupVersion
       }
 
+    inclPPUs :: WhetherToGeneratePPUs
+    inclPPUs =
+        -- We don't generate any other updates, since doing so might
+        -- accidentally supplant the bespoke update that these tests are
+        -- expecting.
+        --
+        -- The transaction this test introduces causes all nodes to propose the
+        -- same parameter update. It'd technically be OK if some nodes then
+        -- changed their proposal to a different update, as long as at least
+        -- @Quorum@-many nodes were still proposing this test's original update
```

Comment:
`sgUpdateQuorum` is currently 1, but since we're asserting that the genesis config must be valid, it trips over the recently introduced https://github.com/input-output-hk/cardano-ledger-specs/blob/c453fa6fce99ed3cb09d129c829ddbf0980a642c/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Genesis.hs#L303, which I fix here https://github.com/input-output-hk/ouroboros-network/pull/2456/commits/54c71ce0836016d7309494984796204064101715. This is fine because all your nodes propose the same update, correct?

Reply:
Yep, the transaction associate the same proposal to every genesis key hash. And we're preventing the random generator from changing any of those associations by preventing it from introducing any updates.